src/ad-products/templates/hide-on-viewability.ts

```diff
@@ -47,24 +49,46 @@ export class HideOnViewability {
 	init(): void {
 		utils.logger(HideOnViewability.getName(), 'init');
 
-		this.registerViewabilityListener(this.adSlot);
+		if (!this.config || !this.config.hideAnimation) {
+			return;
+		}
+
+		Promise.race([this.registerViewabilityListener(this.adSlot), this.registerTimeoutHide()]).then(
+			() => {
+				this.hideSlot(this.adSlot);
+			},
+		);
+	}
+
+	private registerViewabilityListener(adSlot: AdSlot): Promise<void> {
+		return new Promise((resolve) => {
```

Comment:
```suggestion
		return new Promise((resolve) => {
			adSlot.viewed.then(() => {
				setTimeout(() => {
					utils.logger(HideOnViewability.getName(), 'viewability Promise resolved');
					resolve();
				}, this.config.additionalHideTime || 0);
			});
		});
```

Would it work? What do you think? Worth changing? Your call - I'm fine with leaving it as it is too.

Reply:
Yeah, why not. It's still working and it's more readable. Promises FTW!